File: php/observer/ForecastDisplay.php
<?
include_once("WeatherData.php");
include_once("Observer.php");
include_once("DisplayElement.php");

class ForecastDisplay implements Observer, DisplayElement {
    private $currentPressure;
    private $lastPressure;
    private $weatherData;

    public function __construct(WeatherData $weatherData) {
        $this->weatherData = $weatherData;
        $this->weatherData->registerObserver($this);
    }

    public function update ($temp, $humidity, $pressure) {
        $this->lastPressure = $this->currentPressure;
        $this->currentPressure = $pressure;
        $this->display();
    }

    public function display() {
        echo "Forecast: ";
        if($this->currentPressure > $this->lastPressure) {
            echo "Improving weather on the way!\n";
        } elseif ($this->currentPressure == $this->lastPressure) {
            echo "More of the same\n";
        } elseif ($this->currentPressure < $this->lastPressure) {
            echo "Watch out for cooler, rainy weather\n";
        }
    }
}
?>

File: python/2.7/observer/ForecastDisplay.py
from Observer import *
from DisplayElement import *

class ForecastDisplay(Observer, DisplayElement):
    def __init__(self, weatherData):
        self.currentPressure = 29.92
        self.lastPressure = 0.0
        self.weatherData = weatherData
        self.weatherData.registerObserver(self)

    def update(self, temp, humidity, pressure):
        self.lastPressure = self.currentPressure
        self.currentPressure = pressure;
        self.display()

    def display(self):
        print "Forecast: "
        if(self.currentPressure > self.lastPressure):
            print "Improving weather on the way!"
        elif(self.currentPressure == self.lastPressure):
            print "More of the same"
        elif(self.currentPressure < self.lastPressure):
            print "Watch out for cooler, rainy weather"